Framework/Algorithms/test/CalculateIqtTest.h

```diff
@@ -0,0 +1,96 @@
+#ifndef CALCULATEIQTTEST_H_
+#define CALCULATEIQTTEST_H_
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAlgorithms/CalculateIqt.h"
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+
+class CalculateIqtTest : public CxxTest::TestSuite {
+public:
+  static CalculateIqtTest *createSuite() { return new CalculateIqtTest(); }
+  static void destroySuite(CalculateIqtTest *suite) { delete suite; }
+  MatrixWorkspace_sptr m_sampleWorkspace;
+  MatrixWorkspace_sptr m_resolutionWorkspace;
+
+  CalculateIqtTest() {
+    m_sampleWorkspace = setUpWorkspace(true);
+    m_resolutionWorkspace = setUpWorkspace(false);
+  }
+
+  void test_execSuccess() {
```

Comment:
I would consider splitting this up into a few tests to make them more unit like. 

- Check the algorithm executes
- Check the number of histograms is correct
- Check the blocksize is correct
- Check the actual output values are what you expect for a 'physical' example using numbers which are representative of what might be used in reality.